storage/mysql/mysql.go

```diff
@@ -99,17 +99,44 @@ func (s *Storage) writeCheckpoint(ctx context.Context, rawCheckpoint []byte) err
 }
 
 // ReadTile returns a full tile or a partial tile at the given level and index.
-func (s *Storage) ReadTile(ctx context.Context, level, index uint64) ([]byte, error) {
+//
+// TODO: Handle the following scenarios:
+// 1. Full tile request with full tile output: Return full tile.
+// 2. Full tile request with partial tile output: Return error.
+// 3. Partial tile request with full/larger partial tile output: Return trimmed partial tile with correct tile width.
+// 4. Partial tile request with partial tile (same width) output: Return partial tile.
+// 5. Partial tile request with smaller partial tile output: Return error.
+func (s *Storage) ReadTile(ctx context.Context, level, index, width uint64) ([]byte, error) {
```

Comment:
One thing to note is that most of the other code (e.g. the get tiles function passed into `TreeBuilder`) uses `treeSize` rather than width here, but feel free to adjust/adapt to that when you're ready to integrate (haha!) with that.

Reply:
Good to know. This method will probably be updated when the sequencing and integrate flows are added.